lib/saml_idp/request.rb

```diff
@@ -115,10 +115,11 @@ def valid_signature?
     end
 
     def service_provider?
-      service_provider.valid?
+      service_provider && service_provider.valid?
```

Comment:
I would also remove service_provide.valid? completely since it doesn't do anything useful.